src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReplacementService.cs

```diff
@@ -1063,6 +1305,11 @@ await CheckFileExists(releaseId, replacementFileId)
                 .OnSuccessVoid(_cacheService.DeleteItem);
         }
 
+        private static bool IsTotal(string input)
```

Comment:
Yep, you are correct, we've got:
```
        private static bool IsTotal(string input)
        {
            return input.Equals("Total", StringComparison.OrdinalIgnoreCase);
        }
```

and in `Data.Services.FilterAndIndicatorViewModelBuilders` we've got:

```
    private static bool IsLabelTotal<T>(T input, Func<T, string> labelSelector)
    {
        return labelSelector(input).Equals("Total", StringComparison.OrdinalIgnoreCase);
    }
```

I've just made a slight change to inline two usages of it, getting rid of methods `IsFilterItemTotal` and `IsFilterGroupTotal`.

One obvious choice to attempt getting rid of the duplication would be adding an `IsTotal`  to the `FilterGroup` and `FilterItem` types.

That would clean up the 4 usages across the two classes but wouldn't help with `OrderByLabelWithTotalFirst` which uses `IsLabelTotal` with the `labelSelector`. That's used by `OrderByLabelWithTotalFirst` which can be used to order filters or indicator groups so it's got a generic way of getting the label using the selector.

To keep that generic we might need at least `FilterGroup` and `FilterItem` to derive from the same base class with an `IsTotal` method?

I'm wary of adding an `IsTotal` method to `FilterItem` because there's logic around when it gets treated as the 'total' option by the front-end, and it could be a bit misleading.

Regardless of a filter item having the 'total' label, it's not the filter item which gets sets as the `totalFilterItemId` for a filter in the `FilterMetaViewModel` unless it exists within the `Total` group or there's only one group.

Reply:
Nice one, just thought I'd seen it in passing.